torch/_functorch/aot_autograd.py

```diff
@@ -2115,6 +2136,31 @@ def compute_overlapping_inputs(fwd_inputs, aliased_input_indices):
                 actual_aliased_indices.add(j_)
     return actual_aliased_indices
 
+
+def _check_can_be_in_graph(mutates_data, mutates_metadata, requires_grad, keep_input_mutations: bool, is_train: bool):
+    if keep_input_mutations and is_train:
+        return mutates_data and not mutates_metadata and not requires_grad
+    if keep_input_mutations:
+        return mutates_data and not mutates_metadata
+    return False
+
+
+def _check_if_mutation_can_be_handled_in_fw_graph(input_info: InputAliasInfo, keep_input_mutations: bool, is_train: bool):
+    return _check_can_be_in_graph(
+        input_info.mutates_data,
+        input_info.mutates_metadata,
+        input_info.requires_grad,
+        keep_input_mutations,
+        is_train
+    )
+
+
+def _check_if_mutation_will_be_returned_in_runtime(input_info: InputAliasInfo, keep_input_mutations: bool, is_train: bool):
+    return (
+        (input_info.mutates_data or input_info.mutates_metadata) and
```

Comment:
Why do I need this condition?

Reply:
It's because there's a ternary condition going on. In that case, we should have a single function that returns a three-valued enum (MUTATION_INSIDE_GRAPH, MUTATION_OUTSIDE_GRAPH, NO_MUTATION)